packages/devtools_app/lib/src/flutter/common_widgets.dart

```diff
@@ -113,3 +116,163 @@ class DefaultTaggedText extends StatelessWidget {
+  ///
+  /// Visible to grab it in tests.
+  @visibleForTesting
+  Key get dividerKey => Key('$this dividerKey');
+
+  /// The size of the divider between [firstChild] and [secondChild] in
+  /// logical pixels (dp, not px).
+  static const double dividerMainAxisSize = 10.0;
+
+  @override
+  State<StatefulWidget> createState() => _SplitState();
+}
+
+class _SplitState extends State<Split> {
+  double firstFraction;
+  double get secondFraction => 1 - firstFraction;
+  bool get isHorizontal => widget.axis == Axis.horizontal;
+
+  @override
+  void initState() {
+    super.initState();
+    firstFraction = widget.initialFirstFraction;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return LayoutBuilder(builder: _buildLayout);
+  }
+
+  Widget _buildLayout(BuildContext context, BoxConstraints constraints) {
+    final width = constraints.maxWidth;
+    final height = constraints.maxHeight;
+    final axisSize = isHorizontal ? width : height;
+    const halfDivider = Split.dividerMainAxisSize / 2.0;
+    // The fraction of the layout the divider needs to take up from each child.
+    final halfDividerFraction = halfDivider / axisSize;
+
+    final sanitizedFirstFraction =
+        firstFraction.clamp(halfDividerFraction, 1.0 - halfDividerFraction);
+    final sanitizedSecondFraction =
+        secondFraction.clamp(halfDividerFraction, 1.0 - halfDividerFraction);
+
+    final firstSize = axisSize * sanitizedFirstFraction - halfDivider;
+    final secondSize = axisSize * sanitizedSecondFraction - halfDivider;
+
+    void updateSpacing(DragUpdateDetails dragDetails) {
+      final delta = isHorizontal ? dragDetails.delta.dx : dragDetails.delta.dy;
+      final fractionalDelta = delta / axisSize;
+      setState(() {
+        // Update the fraction of space consumed by the children,
+        // being sure not to allocate any of them negative space.
+        firstFraction += fractionalDelta;
+        firstFraction = firstFraction.clamp(0.0, 1.0);
+      });
+    }
+
+    final children = [
+      SizedBox(
+        width: isHorizontal ? firstSize : width,
+        height: isHorizontal ? height : firstSize,
+        child: widget.firstChild,
+      ),
+      GestureDetector(
+        key: widget.dividerKey,
+        onHorizontalDragUpdate: isHorizontal ? updateSpacing : null,
+        onVerticalDragUpdate: isHorizontal ? null : updateSpacing,
+        child: SizedBox(
+          width: isHorizontal ? Split.dividerMainAxisSize : width,
+          height: isHorizontal ? height : Split.dividerMainAxisSize,
+          child: const Center(
+            child: Text(
+              ':::::::',
+              textAlign: TextAlign.center,
+            ),
+          ),
+        ),
+      ),
+      SizedBox(
+        width: isHorizontal ? secondSize : width,
+        height: isHorizontal ? height : secondSize,
+        child: widget.secondChild,
+      ),
+    ];
+    if (widget.axis == Axis.horizontal) {
```

Comment:
this can be replaced with
`Flex(direction: widget.axis, children: children)`

Column and Row are just sugar around Flex with a fixed direction.

Reply:
Good suggestion, done.